plone/app/portlets/tests/test_formextender.py

```diff
@@ -0,0 +1,270 @@
+# -*- coding: utf-8 -*-
+
+from plone.app.portlets.browser.interfaces import (IPortletAddForm,
+                                                   IPortletEditForm)
+from plone.app.portlets.portlets import news
+from plone.app.portlets.storage import PortletAssignmentMapping
+from plone.app.portlets.tests.base import PortletsTestCase
+from plone.portlets.interfaces import (IPortletAssignment,
+                                       IPortletAssignmentSettings,
+                                       IPortletManager, IPortletRenderer,
+                                       IPortletType)
+from plone.z3cform.fieldsets.extensible import FormExtender
+from plone.z3cform.fieldsets.interfaces import IFormExtender
+from z3c.form import field
+from zope import schema
+from zope.component import (adapts, getGlobalSiteManager, getMultiAdapter,
+                            getUtility)
+from zope.interface import Interface, implements
+from zope.publisher.interfaces.browser import IDefaultBrowserLayer
+
+# A sample schemaextender:
+
+
+EXTENDER_PREFIX = 'portletcssclass'
+
+
+class IPortletCssClass(Interface):
+    """ Schema for portlet css class  """
+
+    # css_class is just an example.
+    # In real life a css_class implementation would be a
+    # Choice field with a vocabulary, editable in a controlpanel.
+    css_class = schema.TextLine(
+        title=u'Portlet appearance',
+        required=False
+    )
+
+
+class PortletCssClassFormExtender(FormExtender):
+
+    def update(self):
+        self.add(IPortletCssClass, prefix=EXTENDER_PREFIX)
+
+
+class PortletCssClassAdapter(object):
+    adapts(IPortletAssignment)
+    implements(IPortletCssClass)
```

Comment:
please use the class decorators @adapter and @implementer

Reply:
@jensens done